### DISCUSSION TITLE
Is it possible to manage users from native laravel users table in Twill CMS?

### DISCUSSION THREAD

**Discussion by khrelian** [area17/twill#1279] [Q&A] [answered]
Hey guys, I just discovered twill for laravel and my mind is blown. So far I have been able to follow the tutorials you guys put up in youtube and I'm very amazed by how much effort and ingenuity that must have come into this project.

My dev lead introduced me to this. So far I've been able to do my tasks with ease, but I'm stuck with how I should approach this new requirement. Is it possible to use your modules structure so I can manage the users table from laravel on CMS front end? Is there any documentation that exists somewhere that I could read in order for me to replicate your models and file structure the same way when I use your CRUD modules? 

Please forgive my bad english, it's not my first language. If there is something wrong with my question, or if it's not comprehensible, please let me know so I can clarify.

**Comment by pboivin** [comment, marked as answer]
Hi @khrelian, this is definitely possible. I don't know of a full example of this but essentially, you would need to create a `Users` module and swap out the generated model with Laravel's own `User` model.

Not too long ago, I had a similar requirement for a project. I needed to create a `Profiles` module and automatically attach a profile to new user records. I created a short guide for that use case:
https://gist.github.com/pboivin/d20d23ae130ea9a112ee49dd98f00239

Hope this can help!

**Reply by khrelian** [reply]
Hi @pboivin. Thank you so much for responding. I managed to create a module for laravel's user table by checking out your code from vendor. 

I managed to make everything work, except that when I try to add a browser to my form to link a company_id field, the list returns blank, when I already populated my companies table. 

I tried overriding the browser method in my Company controller. I seem to have managed to get the list working, but the company_id won't save when I try to update a user's record.

It has been so much fun exploring how you guys engineered this, and I'd like to explore a little bit further. Unfortunately, time is a huge factor on this requirement. I'll take a look at your link though, and see if it helps. Thanks again!

**Comment by khrelian** [comment]
For anyone else trying to create a module that will manage users from laravel's native users table. Hope this helps.

- Edit users table  and add a `deleted_at` field
- Create `Controllers/Admin/UsersManagerController`. This should extend Twill's module controller
- set the following:
> protected $moduleName = 'UsersManager'
> protected $modelName = 'LaraUser'
- Create 'LaraUsersRequest' (for twill's CRUD to work)
- Create the model. In this case, we'll name it 'LaraUser' and extend Twill's main model A17\Twill\Models\Model
- Create the repository 'LaraUserRepository'. This should extend Twill's ModuleRepository. Add the necessary __construct as needed.
- Create the views
> LaraUsers/form.blade.php
> LaraUsers/index.blade.php
- create the routes in your route file 'Route::module('usersManager')'

**Reply by pboivin** [reply]
Hi @khrelian, this is good to have for reference, thanks! In this example, how does the `LaraUser` model get associated to the built-in `users` table?

**Reply by khrelian** [reply]
hi @pboivin, good catch! I forgot to mention that you should set the protected $table property to `users`
